pkgs/jnigen/test/simple_package_test/runtime_test_registrant.dart

```diff
@@ -687,6 +687,29 @@ void registerTests(String groupName, TestRunnerCallback test) {
         }
       }
     });
+
+    group('throw Java exceptions', () {
+      test('StringConverter.implement', () async {
+        final stringConverter = StringConverter.implement($StringConverterImpl(
+          parseToInt: (s) {
+            final firstChar = s.toDartString().substring(0, 1);
+            final value = int.tryParse(firstChar);
+            if (value == null || value > 5) {
```

Comment:
nit: This is a weird logic, why should `parseToInt` fail if `value > 5`? Why do we only parse the first digit?

I think it's better to just go with `tryParse` and only throw when that's null. Your test could be trying to convert `foo` (and failing, so -1) and `700` (and succeeding). It's easier to understand the test this way and it achieves the same thing.

Reply:
Done ✅ 

I was actually exploring throwing multiple exceptions and only handling a select few in the Java code, but that turned out to be troublesome and I had to remove the other custom exceptions.